include/matx/core/make_sparse_tensor.h

```diff
@@ -59,10 +59,15 @@ __MATX_INLINE__ static auto makeDefaultNonOwningStorage() {
 template <typename ValTensor, typename CrdTensor>
 auto make_tensor_coo(ValTensor &val, CrdTensor &row, CrdTensor &col,
                      const index_t (&shape)[2]) {
-  static_assert(ValTensor::Rank() == 1 && CrdTensor::Rank() == 1);
   using VAL = typename ValTensor::value_type;
   using CRD = typename CrdTensor::value_type;
   using POS = index_t;
+  // Proper structure.
+  MATX_ASSERT_STR(val.Rank() == 1 && row.Rank() == 1 && col.Rank() == 1,
```

Comment:
MATX_STATIC_ASSERT_STR or just static_assert

Reply:
is there a preference one way or the other? what would you recommend?